### ISSUE TITLE
lesspipe: missing dependency on file 

### ISSUE THREAD

**Issue by Moredread** [NixOS/nixpkgs#37709]
## Issue description

On my system when using lesspipe I get a message complaining that ` file` is missing

```
/nix/store/cfrkqx4cjm6hj0szgh22q6yhrvglnljf-lesspipe-1.82/bin/lesspipe.sh: line 47: file: command not found
```

The original PR 15338 acknowledges that dependencies aren't handled. As lesspipe makes use of what you have installed on your system, it might be too much to pull in tools for all the possible supported file formats.

At least having `file` as a dependency seems to make sense though, as it is needed for running in each case.

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.9.85, NixOS, 17.09.3129.1dcd022f01b (Hummingbird)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0pre5889_c287d731`
 - channels(root): `"nixos-17.09, nixos-unstable"`
 - channels(user): `"nixos-17.09.3200.f4f3e19eac8, nixos-unstable-18.03pre130932.cc4677c36ee"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by Moredread** [comment]
@snaar already fixed it in e0d5f644ac0b63570f5aa989072e71312cadbb5b

Thanks!

**Comment by snaar** [comment]
For completeness, it was also in https://github.com/NixOS/nixpkgs/commit/c6bc77d30e39cc345f36d0c33979819dcb305d2a and https://github.com/NixOS/nixpkgs/commit/93207805b8e1332f45d49e94010a4d5803166c6e
